README.md

```diff
@@ -87,6 +89,21 @@ Whenever the server obtains a connection on the remote port, it generates a secu
 
 For correctness reasons and to avoid memory leaks, incoming connections are only stored by the server for up to 10 seconds before being discarded if the client does not accept them.
 
+## Optional authentication
+
+The client and server can optionally use secrets to protect the server from
+being used by others. The secret is sent only by the client as encrypted base64
+by using itself as a key with a randomly generated nonce. No additional traffic
+is encrypted; only the secret.
```

Comment:
What benefit does encrypting the secret provide here? It seems like any client could replay the same encrypted secret + nonce combination, so this doesn't provide any security benefit beyond a plaintext exchange of secrets.

Reply:
Absolutely, I forgot to consider replay attacks. I'll add the server challenge to make encryption actually work properly.